validator/src/scheduler/execution_result_ffi.rs

```diff
@@ -47,11 +47,13 @@ impl<'source> FromPyObject<'source> for BatchResult {
     fn extract(py: cpython::Python, obj: &'source cpython::PyObject) -> cpython::PyResult<Self> {
         let state_hash = obj.getattr(py, "state_hash").unwrap();
 
-        let sh = state_hash.extract::<String>(py)?;
+        let sh = if state_hash != cpython::Python::None(py) {
```

Comment:
`FromPyObject` is implemented for `Option<T: FromPyObject>`.  This could just be:
```
let sh: Option<String> = cpython::extract(py)?;
```

Reply:
Cool. I hadn't known that. I will change this and reevaluate my use of checking Python::None(py).